monix-eval/shared/src/main/scala/monix/eval/TaskLike.scala

```diff
@@ -0,0 +1,163 @@
+  * where the internals are specialized to `Task` anyway, like for
+  * example the implementation of `monix.reactive.Observable`.
+  */
+@implicitNotFound("""Cannot find implicit value for TaskLike[${F}].
+Building this implicit value might depend on having an implicit
+s.c.ExecutionContext in scope, a Scheduler or some equivalent type.""")
+trait TaskLike[F[_]] {
+  /**
+    * `ToTask` has a `Monad` restriction, expressed via composition
+    * instead of inheritance.
+    */
+  def monad: Monad[F]
+
+  /**
+    * Converts from `F[A]` to `Task[A]`, preserving referential
+    * transparency if `F[_]` is a pure data type and preserving
+    * interruptibility if the source is cancelable.
+    */
+  def toTask[A](fa: F[A]): Task[A]
+}
+
+object TaskLike extends ToTaskImplicits0 {
+  /**
+    * Returns the available instance for `F`.
+    */
+  def apply[F[_]](implicit F: TaskLike[F]): TaskLike[F] = F
+
+  /**
+    * Instance for `Task`, returning same reference.
+    */
+  implicit val fromTask: TaskLike[Task] =
+    new TaskLike[Task] {
+      val monad = Monad[Task]
+      def toTask[A](fa: Task[A]): Task[A] = fa
+    }
+
+  /**
+    * Converts to `Task` from [[scala.concurrent.Future]].
+    */
+  implicit def fromFuture(implicit ec: ExecutionContext): TaskLike[Future] =
+    new TaskLike[Future] {
+      val monad = catsStdInstancesForFuture
+      def toTask[A](fa: Future[A]): Task[A] =
+        Task.fromFuture(fa)
+    }
+
+  /**
+    * Converts to `Task` from [[Coeval]].
+    */
+  implicit val fromCoeval: TaskLike[Coeval] =
+    new TaskLike[Coeval] {
+      val monad = Monad[Coeval]
+      def toTask[A](fa: Coeval[A]): Task[A] =
+        Task.coeval(fa)
+    }
+
+  /**
+    * Converts to `Task` from [[Coeval]].
+    */
+  implicit val fromEval: TaskLike[Eval] =
+    new TaskLike[Eval] {
+      val monad = Monad[Eval]
+      def toTask[A](fa: Eval[A]): Task[A] =
+        Task.fromEval(fa)
+    }
+
+  /**
+    * Converts to `Task` from
+    * [[https://typelevel.org/cats-effect/datatypes/io.html cats.effect.IO]].
+    */
+  implicit val fromIO: TaskLike[IO] =
+    new TaskLike[IO] {
+      val monad = Monad[IO]
+      def toTask[A](fa: IO[A]): Task[A] =
+        Task.fromIO(fa)
+    }
+}
+
+private[eval] abstract class ToTaskImplicits0 extends ToTaskImplicits1 {
+  /**
+    * Converts to `Task` from
+    * [[https://typelevel.org/cats-effect/typeclasses/concurrent-effect.html cats.effect.ConcurrentEffect]].
+    */
+  implicit def fromConcurrent[F[_]](implicit F: ConcurrentEffect[F]): TaskLike[F] =
+    new TaskLike[F] {
+      val monad = F
+      def toTask[A](fa: F[A]): Task[A] =
+        Task.fromConcurrent(fa)
+    }
+}
+
+private[eval] abstract class ToTaskImplicits1 {
+  /**
+    * Converts to `Task` from
+    * [[https://typelevel.org/cats-effect/typeclasses/concurrent-effect.html cats.effect.Async]].
+    */
+  implicit def fromAsync[F[_]](implicit F: Effect[F]): TaskLike[F] =
```

Comment:
I know it's internal that won't be written explicitly, but maybe it should be named `fromEffect`? 😄